### ISSUE TITLE
`ENOENT: no such file or directory` during Vercel deployment

### ISSUE THREAD

**Issue by binary64** [nextauthjs/next-auth#8050]
### Environment

This from Vercel, which is set to use Node 18.

```
07:19:28.149 | OS: Linux 5.10 Amazon Linux 2
07:19:28.149 | CPU: (4) x64 Intel(R) Xeon(R) Platinum 8259CL CPU @ 2.50GHz
07:19:28.149 | Memory: 14.29 GB / 15.32 GB
07:19:28.149 | Container: Yes
07:19:28.149 | Shell: 4.2.46 - /bin/bash
07:19:28.149 | Binaries:
07:19:28.149 | Node: 18.16.1 - /node18/bin/node
07:19:28.149 | Yarn: 1.22.17 - ~/.yarn/bin/yarn
07:19:28.150 | npm: 9.5.1 - /node18/bin/npm
07:19:28.150 | pnpm: 6.35.1 - /usr/local/bin/pnpm
```

### Reproduction URL

https://github.com/binary64/blabber/actions/runs/5566184175/jobs/10167098969

### Describe the issue

I have created a very simple reproduction app, which uses Nx to deploy a NextJS app to Vercel using the Build API. This means Vercel don't have to have access to my whole repo in order to deploy.

However, I hit this mystery error:
```
Error: ENOENT: no such file or directory, lstat '/node_modules/.pnpm/next@13.3.4_@babel+core@7.21.8_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/server/next-server.js'
```

### How to reproduce

1. Create a new Nx repo, add a nextjs app into it (myapp), add next-auth into that
2. yarn nx run myapp:build --prod
3. yarn nx run myapp:export
4. cd dist/apps/myapp
5. vercel --prod

Or inspect my CI process. If you see the commit before I added next-auth, everything was working.

### Expected behavior

The deployed Vercel app should function the same as it does locally, which is fine.

**Comment by balazsorban44** [comment]
Wrong repo, I could not see `next-auth`, but it would be likely irrelevant anyway. I recommend opening a support ticket at Vercel if the issue persists.

**Comment by binary64** [comment]
Apologies, the GHA URL I linked to was not the clearest. It was just adding the npx envinfo command your Issues recommended. The commit URL I think you were more expecting is this one: https://github.com/binary64/blabber/commit/1cc38a333aba29ac130272d9a5abf29dcf5d4d41 which has next-auth in the modified yarn.lock. Sorry I would usually squash this in advance, but in this case I am practicing some Trunk-based development.

In summary I still feel this is a next-auth issue, demonstrated by CI going green the moment I reverted the above commit. I do admit the error doesn't suggest it is next-auth, but because of my atomic commit above, it can't be much else (that I can currently think of!).

I appriciate your help.
